src/memory/buffer_pool.cpp

```diff
@@ -6,30 +6,95 @@ SMILE_NS_BEGIN
 
 BufferPool::BufferPool( FileStorage* storage, const BufferPoolConfig& config ) noexcept {
 	p_storage = storage;
-	p_pool = (char*) malloc( config.m_poolSizeKB*1024*sizeof(char) );
+	m_config = config;	
+}
+
+ErrorCode BufferPool::open( const std::string& path ) noexcept {
+	ErrorCode error = p_storage->open(path);
```

Comment:
If the storage is opened from inside the buffer pool, then it should not be passed as a parameter when constructing the buffer pool object. In otherwords, either build a buffer pool with an already opened storage or open the storage inside the open method of the buffer pool. I'd go the second option.

Reply:
Fixed in 752c787.